server/lib/chat/chat-service.ts

```diff
@@ -309,21 +313,21 @@ export default class ChatService {
           throw err
         }
       }
-    } while (!succeeded && !isUserBanned && attempts < MAX_JOIN_ATTEMPTS)
+    } while (!succeeded && !exitCode && attempts < MAX_JOIN_ATTEMPTS)
 
-    if (maximumJoinedChannels) {
+    if (exitCode === JoinChannelExitCode.MaximumJoinedChannels) {
       throw new ChatServiceError(
         ChatServiceErrorCode.MaximumJoinedChannels,
         'Maximum joined channels reached',
       )
     }
-    if (maximumOwnedChannels) {
+    if (exitCode === JoinChannelExitCode.MaximumOwnedChannels) {
       throw new ChatServiceError(
         ChatServiceErrorCode.MaximumOwnedChannels,
         'Maximum owned channels reached',
       )
     }
-    if (isUserBanned) {
+    if (exitCode === JoinChannelExitCode.UserBanned) {
```

Comment:
You should add an `assertUnreachable` clause after these to make sure these all get checked:
```ts
if (exitCode !== undefined) {
  assertUnreachable(exitCode)
}
```